### PR TITLE
Therapist Profile

### PR DESCRIPTION
**Pull request by MohammadAlhalaq**

### CODE REVIEW

File: client/src/components/common/AvailabilityTime/style.css [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+}
+
+.Availabile_table {
+  width: max-content;
+  border-radius: 1rem; 
+}
+
+.ant-table td, .ant-table th{
+  width: 6.7rem;
+}
```
Comment (fadeomar) [inline client/src/components/common/AvailabilityTime/style.css]: new line :+1:

File: client/src/components/common/CardProfile/index.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        <Avatar shape="circle" src={imgUrl} size={100} icon="user" />
+
+        <section className="profile_card_b">
+          <h4>Approach: {therapyApproach}</h4>
+          <h4>Type: {therapyType}</h4>
+        </section>
+      </section>
+      <section className="profile_card_fees">
+        <h3>
+          <Icon type="dollar" />
```
Comment (fadeomar) [inline client/src/components/common/CardProfile/index.js]: you can use ( shift + 4 ) droller sing in you keyboard , review the figma it's not icon , it symbol

File: client/src/components/common/CardProfile/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+const Card = ({ cardInfo: { imgUrl, therapyApproach, therapyType, fees } }) => {
+  return (
+    <section className="profile_card_container">
+      <section className="profile_card">
+        <Avatar shape="circle" src={imgUrl} size={100} icon="user" />
+
+        <section className="profile_card_b">
+          <h4>Approach: {therapyApproach}</h4>
+          <h4>Type: {therapyType}</h4>
```
Comment (fadeomar) [inline client/src/components/common/CardProfile/index.js]: NOTE: The section element is not a generic container element. When an element is needed for styling purposes or as a convenience for scripting, authors are encouraged to use the div element instead. A general rule is that the section element is appropriate only if the element's contents would be listed explicitly in the document's outline.

[https://stackoverflow.com/questions/7041283/nesting-html5-section-tags](stackoverflow)

File: client/src/components/common/CardProfile/index.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { Icon, Avatar } from 'antd';
+import propTypes from 'prop-types';
+
+import './style.css';
+
+const Card = ({ cardInfo: { imgUrl, therapyApproach, therapyType, fees } }) => {
```
Comment (fadeomar) [inline client/src/components/common/CardProfile/index.js]: ```` fees ===> priceRange````

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export default table => {
+  return table.filter(item => {
+    return item.day.length >= 3 && item.from.length > 3 && item.to.length > 3;
```
Comment (fadeomar) [inline client/src/components/common/AvailabilityTime/filterAvailableTime.js]: I think every item in the table array is an object not an array and you cant use filter on object nether length .
you can run this component in see what would you got 
[https://github.com/GSG-G7/shrinc/blob/master/client/src/components/common/availabilityTable/index.js](url)

File: client/src/components/common/CardProfile/index.js (hunk 2 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/CardProfile/index.js]: We don't want background for icon

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js (hunk 5 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/filterAvailableTime.js]: the table is array 
```js
available: [
    { key: 1, day: 'Mon', from: '12:00 am', to: '1:00 pm' },
    { key: 2, day: 'Tue', from: '', to: '' },
    { key: 3, day: 'Wen', from: '', to: '' },
    { key: 4, day: 'Thu', from: '', to: '' },
    { key: 6, day: 'Fri', from: '', to: '' },
    { key: 7, day: 'Sat', from: '1:00 pm', to: '2:00 pm' },
    { key: 8, day: 'Sun', from: '', to: '' },
  ]
```

File: client/src/components/common/CardProfile/index.js (hunk 4 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/CardProfile/index.js]: done

File: client/src/components/common/AvailabilityTime/style.css (hunk 1 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/style.css]: done

File: client/src/components/common/AvailabilityTime/index.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Icon, Table } from 'antd';
+import propTypes from 'prop-types';
+import func from './filterAvailableTime';
```
Comment (hashemt) [inline client/src/components/common/AvailabilityTime/index.js]: add line

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+export default table => {
+  return table.filter(item => {
```
Comment (fadeomar) [inline client/src/components/common/AvailabilityTime/filterAvailableTime.js]:
```suggestion
  return table.filter(item =>  item.from && item.to )
```

File: client/src/components/common/Location/index.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import propTypes from 'prop-types';
+
```
Comment (fadeomar) [inline client/src/components/common/Location/index.js]: delete this line

File: client/src/components/common/Location/style.css [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+  margin-right: 1rem;
+}
+
+.location_img_map {
+  display: block;
+  width: 344px;
+  height: 126px;
+  margin-left: 1rem;
+  margin-bottom: 1rem;
+}
```
Comment (fadeomar) [inline client/src/components/common/Location/style.css]: line here

File: client/src/components/common/Location/index.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+      <section className="location_contact_container">
+        <h2>
+          <Icon type="global" /> Location
+        </h2>
+        <section className="location_contact_info">
+          <section className="location_info">
+            <h5 className="location_titles">Address</h5>
+            <p>{address}</p>
+          </section>
+          <section className="location_info">
```
Comment (fadeomar) [inline client/src/components/common/Location/index.js]:
```suggestion
          <div className="location_info">
```

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js [hunk 11]
```diff
@@ -0,0 +1 @@
+export default table => table.filter(item => item.from && item.to);
```
Comment (Asmaathabet) [inline client/src/components/common/AvailabilityTime/filterAvailableTime.js]: this isn't needed a file for this function , just remove it

File: client/src/components/common/AvailabilityTime/index.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { Icon, Table } from 'antd';
+import propTypes from 'prop-types';
```
Comment (Asmaathabet) [inline client/src/components/common/AvailabilityTime/index.js]: add line here

File: client/src/components/common/Location/index.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+const Location = ({ locationInfo: { address, postalCode, map } }) => {
+  return (
+    <section>
+      <section className="location_contact_container">
+        <h2>
+          <Icon type="global" /> Location
+        </h2>
+        <section className="location_contact_info">
+          <section className="location_info">
```
Comment (fadeomar) [inline client/src/components/common/Location/index.js]:
```suggestion
          <div className="location_info">
```

File: client/src/components/pages/Profile/index.js [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    available,
+    contactinfo: { email, phone, skype },
+    locationInfo,
+  },
+}) => {
+  return (
+    <section className="font">
+      {/* 
+      -- card 
+      */}
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: remove these lines

File: client/src/components/common/AvailabilityTime/index.js [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+        key="table"
+        dataSource={tableData}
+        columns={columns}
+        className="Availabile_table"
+      />
+    </section>
+  );
+};
+
+export default Available;
```
Comment (hashemt) [inline client/src/components/common/AvailabilityTime/index.js]: move export to the end of file

File: client/src/components/pages/Profile/index.js [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+      <section className="profile_card_container">
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: delete commented lines

File: client/src/components/common/AvailabilityTime/style.css [hunk 17]
```diff
@@ -0,0 +1,13 @@
+.Available_contailner {
+  margin:1rem 0 0 2rem;
```
Comment (hashemt) [inline client/src/components/common/AvailabilityTime/style.css]:
```suggestion
  margin: 1rem 0 0 2rem;
```

File: client/src/components/pages/Profile/index.js [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+        <section className="profile_card_fees">
+          <h3>
+            <Icon type="dollar" />
+            &nbsp; &nbsp;
+            <span>Fees {priceRange}$ Session</span>
+          </h3>
+        </section>
+      </section>
+      {/* 
+      -- Available time
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: also here delete commented lines

File: client/src/components/pages/Profile/index.js [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+            <span>Fees {priceRange}$ Session</span>
+          </h3>
+        </section>
+      </section>
+      {/* 
+      -- Available time 
+      */}
+      <AvailableityTime availableityTime={available} />
+      {/* 
+      -- is remote
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: also here delete commented lines

File: client/src/components/pages/Profile/index.js [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+      <section className="profile_contact_container">
+        <h1>
+          <Icon type="wifi" /> Is remotly
+        </h1>
+        <section className="profile_contact_info">
+          <p>{isRemote ? 'Yes' : 'No'}</p>
+        </section>
+      </section>
+      {/** 
+      -- contact
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: also delete commented lines

File: client/src/components/pages/Profile/index.js [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+       */}
+      <section className="profile_contact_container">
+        <h2>
+          <Icon type="phone" /> Contact Info
+        </h2>
+        <section className="profile_contact_info">
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]:
```suggestion
        <div className="profile_contact_info">
```

File: client/src/components/pages/Profile/index.js [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+          <h4>Email</h4>
+          <p>{email}</p>
+          <h4>Phone</h4>
+          <p>{phone}</p>
+          <h4>Skype</h4>
+          <p>{skype}</p>
+        </section>
+      </section>
+      {/** 
+      -- location
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]: delete commented lines

File: client/src/components/pages/Profile/index.js [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+import './style.css';
+
+const Profile = ({
+  profileData: {
+    isRemote,
+    cardInfo: { username, imgUrl, therapyApproach, therapyType, priceRange },
+    available,
+    contactinfo: { email, phone, skype },
+    locationInfo,
+  },
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: you already have a profile data file , so you can delete this and envoke data from the file that has static data .

File: client/src/components/pages/Profile/index.js [hunk 24]
```diff
@@ -0,0 +1,96 @@
...
+      */}
+      <AvailableityTime availableityTime={available} />
+      {/* 
+      -- is remote
+       */}
+      <section className="profile_contact_container">
+        <h1>
+          <Icon type="wifi" /> Is remotly
+        </h1>
+        <section className="profile_contact_info">
```
Comment (fadeomar) [inline client/src/components/pages/Profile/index.js]:
```suggestion
        <div className="profile_contact_info">
```

File: client/src/components/common/Location/index.js (hunk 8 above)
Comment (hashemt) [inline client/src/components/common/Location/index.js]: delete this line

File: client/src/components/pages/Profile/index.js [hunk 25]
```diff
@@ -0,0 +1,96 @@
...
+          <h3>
+            <Icon type="dollar" />
+            &nbsp; &nbsp;
+            <span>Fees {priceRange}$ Session</span>
+          </h3>
+        </section>
+      </section>
+      {/* 
+      -- Available time 
+      */}
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: delete these lines

File: client/src/components/pages/Profile/style.css [hunk 26]
```diff
@@ -0,0 +1,66 @@
+.font {
+  font-family: Roboto, sans-serif;
+}
+
+/* profile card */
```
Comment (fadeomar) [inline client/src/components/pages/Profile/style.css]: really commented line in css

File: client/src/components/pages/Profile/index.js [hunk 27]
```diff
@@ -0,0 +1,96 @@
...
+        <h1>
+          <Icon type="wifi" /> Is remotly
+        </h1>
+        <section className="profile_contact_info">
+          <p>{isRemote ? 'Yes' : 'No'}</p>
+        </section>
+      </section>
+      {/** 
+      -- contact
+       */}
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: delete these line & use comment like this 
//contact

File: client/src/components/pages/Profile/index.js [hunk 28]
```diff
@@ -0,0 +1,96 @@
...
+          </h3>
+        </section>
+      </section>
+      {/* 
+      -- Available time 
+      */}
+      <AvailableityTime availableityTime={available} />
+      {/* 
+      -- is remote
+       */}
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: // is remote

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js (hunk 11 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/filterAvailableTime.js]: done

File: client/src/components/pages/Profile/index.js [hunk 29]
```diff
@@ -0,0 +1,96 @@
...
+          <p>{email}</p>
+          <h4>Phone</h4>
+          <p>{phone}</p>
+          <h4>Skype</h4>
+          <p>{skype}</p>
+        </section>
+      </section>
+      {/** 
+      -- location
+       */}
```
Comment (Asmaathabet) [inline client/src/components/pages/Profile/index.js]: //location

File: client/src/components/pages/Profile/style.css [hunk 30]
```diff
@@ -0,0 +1,66 @@
...
+
+/* contact */
+.profile_contact_container {
+  margin:1rem 0 0 2rem;
+}
+
+.profile_contact_info {
+  color: #32A48F;
+  margin-left: 2rem;
+}
```
Comment (fadeomar) [inline client/src/components/pages/Profile/style.css]: line here

File: client/src/components/common/AvailabilityTime/index.js (hunk 6 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/index.js]: done

File: client/src/components/common/Location/style.css (hunk 9 above)
Comment (hashemt) [inline client/src/components/common/Location/style.css]: add line

File: client/src/components/common/AvailabilityTime/index.js (hunk 6 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/index.js]: done

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js (hunk 11 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/filterAvailableTime.js]: done

File: client/src/components/common/AvailabilityTime/index.js (hunk 12 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/index.js]: Done

File: client/src/components/common/AvailabilityTime/index.js (hunk 15 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/index.js]: Done

File: client/src/components/common/Location/index.js (hunk 8 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/Location/index.js]: Done

File: client/src/components/pages/Profile/style.css [hunk 31]
```diff
@@ -0,0 +1,66 @@
...
+
+.profile_card {
+  display: flex;
+  width: 100%;
+  margin-top: 1rem;
+  justify-content: space-evenly;
+  text-align: center;
+}
+
+.profile_card_a{
```
Comment (hashemt) [inline client/src/components/pages/Profile/style.css]:
```suggestion
.profile_card_a {
```

File: client/src/components/pages/Profile/style.css (hunk 30 above)
Comment (hashemt) [inline client/src/components/pages/Profile/style.css]: add new line

File: client/src/components/common/AvailabilityTime/filterAvailableTime.js (hunk 7 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/AvailabilityTime/filterAvailableTime.js]: Done

File: client/src/components/common/Location/style.css (hunk 9 above)
Comment (MohammadAlhalaq) [reply client/src/components/common/Location/style.css]: Done

File: client/src/components/common/AvailabilityTime/index.js [hunk 32]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Icon, Table } from 'antd';
+import propTypes from 'prop-types';
+
+import './style.css';
+
+const filterTime = table => table.filter(item => item.from && item.to);
+
+const Available = ({ availableityTime }) => {
+  const tableData = filterTime(availableityTime);
```
Comment (hashemt) [inline client/src/components/common/AvailabilityTime/index.js]:
```suggestion
  const tableData = availableityTime.filter(item => item.from && item.to);;
```

**Pull request merged**